assigner.py

```diff
@@ -20,173 +22,257 @@
+        host = conf['gitlab-host']
+        namespace = conf['namespace']
+        token = conf['token']
+
+        if dry_run:
+            url = Repo.build_url(host, namespace, hw_name)
+            print("Created repo at {}.".format(url))
         else:
             try:
-                repo = BaseRepo.new(args.name, conf['namespace'], conf['gitlab-host'], conf['token'])
-                print("Created repo at ", repo.url)
+                repo = BaseRepo.new(hw_name, namespace, host, token)
+                print("Created repo at {}.".format(repo.url))
             except HTTPError as e:
                 if e.response.status_code == 400:
-                    logger.warning("Repository %s already exists!", args.name)
+                    logger.warning("Repository {} already exists!".format(hw_name))
                 else:
                     raise
 
 
 def assign(args):
-    if args.dry_run:
-        raise NotImplementedError("'--dry-run' is not implemented")
-    if args.student:
+    """Creates homework repositories for an assignment for each student
+    in the roster.
+    """
+    hw_name = args.name
+    branch = args.branch
+    section = args.section
+    dry_run = args.dry_run
+    force = args.force
+    verbose = args.verbose
+    target = args.student  # used if assigning to a single student
+
+    if target:
         raise NotImplementedError("'--student' is not implemented")
 
     with config(args.config) as conf, tempfile.TemporaryDirectory() as tmpdirname:
-        base = BaseRepo(conf['gitlab-host'], conf['namespace'], args.name, conf['token'])
-        if args.branch:
-            base.clone_to(tmpdirname, args.branch)
+        host = conf['gitlab-host']
+        namespace = conf['namespace']
+        token = conf['token']
+        semester = conf['semester']
+        if section:
+            roster = [s for s in conf['roster'] if s['section'] != section]
         else:
-            base.clone_to(tmpdirname)
-
-        count = 0
-        for student in conf['roster']:
-            if args.section and student['section'] != args.section:
-                continue
-
+            roster = conf['roster']
+
+        base = BaseRepo(host, namespace, hw_name, token)
+        if not dry_run:
+            base.clone_to(tmpdirname, branch)
+
+        actual_count = 0  # Represents the number of repos actually pushed to
+        s_count = len(roster)
+        print("Assigning {} to {} student{} in {}...".format(
+            hw_name, s_count,
+            "s" if s_count != 1 else "",
+            "section " + section if section else "any section")
+        )
+        for i, student in enumerate(roster):
+            username = student['username']
+            s_section = student['section']
+            full_name = StudentRepo.name(semester, s_section, hw_name, username)
+            repo = StudentRepo(host, namespace, full_name, token)
+
+            # Is it always necessary to log this?  Feels wrong.
```

Comment:
The key bit here is the line `repo.info`. This try-catch attempts to get an already existing repo; if it doesn't exist, `repo.info` will throw and the exception will handle creating the repo. If we get past `repo.info`, then the repo must already exist. 

If you couldn't tell, this feature was added an hour before I needed it for a lab.